src/main/java/pl/matidominati/GitHubApp/service/ServiceImpl.java

```
@@ -0,0 +1,35 @@
+package pl.matidominati.GitHubApp.service;
+
+import feign.FeignException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.matidominati.GitHubApp.client.GitHubFeignClient;
+import pl.matidominati.GitHubApp.client.model.GitHubRepo;
+import pl.matidominati.GitHubApp.exception.DataNotFoundException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class ServiceImpl {
+
+    private final GitHubFeignClient gitHubFeignClient;
+
+    public GitHubRepo getRepositoryDetails(String owner, String repositoryName) {
+        Optional<GitHubRepo> githubRepo = gitHubFeignClient.getRepositoryDetails(owner, repositoryName);
```

Moglibysmy w zgodnie z ideą fail fast -> sprawdzić na poczatku czy np owner lub repository name nie sa puste.